wicker/core/utils.py

```diff
@@ -0,0 +1,14 @@
+import contextlib
+import signal
+
+@contextlib.contextmanager
+def time_limit(seconds):
+    def signal_handler(signum, frame):
+        raise TimeoutError("Timed out!")
```

Comment:
Let's have a more informative message: maybe "Timing out in trying to download file in Wicker"

Reply:
Done.